### PR TITLE
13329: Don't create new GpID's for temporary pieceslots

### PR DESCRIPTION
**Pull request by BrentEaston**
used for editing pieces.

### CODE REVIEW

**Comment by BrentEaston** [comment]
Putting this one hold while I investigate more issues raised by Barbanera.

**Comment by BrentEaston** [comment]
3 related fixes for issue 13329. Ready to go for 3.4.0

File: vassal-app/src/main/java/VASSAL/build/GpIdChecker.java [hunk 1]
```diff
@@ -160,7 +160,7 @@ protected void testGpId(String id, SlotElement element) {
           final int iid = Integer.parseInt(id);
           goodSlots.put(id, element);         // gpid is good.
           if (iid >= maxId) {
-            maxId = iid + 1;
+            maxId = iid;
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/GpIdChecker.java:163]: Couldn't it be `>` instead of `>=`?

**Review by uckelman — CHANGES_REQUESTED** [review]

File: vassal-app/src/main/java/VASSAL/build/GpIdChecker.java (hunk 1 above)
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/build/GpIdChecker.java:163]: Indeed. Fixed

**Review by uckelman — APPROVED** [review]

**Pull request merged**